packages/components/pie-button/test/visual/pie-button.spec.ts

```diff
@@ -14,18 +23,44 @@ const props: PropObject = {
     disabled: [true, false],
 };
 
-const componentPropsMatrix : Combination[] = getAllPropCombinations(props);
-const componentPropsMatrixByVariant: Record<string, Combination[]> = splitCombinationsByPropertyValue(componentPropsMatrix, 'variant');
+// Renders a <pie-button> HTML string with the given prop values
+const renderTestPieButton = (propVals: WebComponentPropValues) => `
+    <pie-button
+        variant="${propVals.variant}"
+        size="${propVals.size}"
+        type="${propVals.type}"
+        ${propVals.isFullWidth ? 'isFullWidth' : ''}
+        ${propVals.disabled ? 'disabled' : ''}>
+        Hello world
+    </pie-button>`;
+
+const componentPropsMatrix : WebComponentPropValues[] = getAllPropCombinations(props);
+const componentPropsMatrixByVariant: Record<string, WebComponentPropValues[]> = splitCombinationsByPropertyValue(componentPropsMatrix, 'variant');
 const componentVariants: string[] = Object.keys(componentPropsMatrixByVariant);
 
+// This ensures the component is registered in the DOM for each test
+// This is not required if your tests mount the web component directly in the tests
+test('Component registered in the DOM', async ({ mount }) => {
```

Comment:
This sounds like a use-case for `beforeEach`?

Reply:
As discussed I'm going to look into a way to do this as simply mounting in `beforeEach` makes an extra component instance visible on the page at the top of the page which isn't ideal